[Term Entry] Kotlin Datetime .atStartOfDayIn() (#3917)
Codecademy/docs: content/kotlin/concepts/datetime/terms/atStartOfDayIn/atStartOfDayIn.md

---
Title: '.atStartOfDayIn()'
Description: 'Finds an instant that corresponds to the start of a date in a particular time zone.'
Subjects:
  - 'Computer Science'
  - 'Code Foundations'
Tags:
  - 'Functions'
  - 'Data Structures'
  - 'Time'
  - 'Date'
CatalogContent:
  - 'learn-kotlin'
  - 'paths/computer-science'
---

The **`.atStartOfDayIn()`** function is used to obtain the start of the day `LocalDateTime` in a specific time zone. It returns an `Instant` (an instantaneous point in time. It is like a super-accurate clock in Kotlin which helps us pinpoint the exact moment when something happens in our code) that corresponds to the start of a date in a particular time zone.

## Syntax

```pseudo
fun LocalDate.atStartOfDayIn(TimeZone): Instant
```

- `LocalDate`: represents a date without reference to a particular time zone.
- `TimeZone`: represents a time zone in which date and time calculations are performed.
- `Instant`: a moment in time.

## Example

The following example demonstrates how to use `.atStartOfDayIn`.

> ```kotlin
> import kotlinx.datetime.*
> 
> fun main() {
>     val todaysDate = LocalDate(2023, 11, 18)
>     val myTimeZone = TimeZone.UTC
>     val startOfToday = todaysDate.atStartOfDayIn(myTimeZone)
>     println("Start of the day November 18th, 2023 in UTC is: $startOfToday")
> }

Reviewer: ```suggestion
```kotlin
import kotlinx.datetime.*

fun main() {
    val todaysDate = LocalDate(2023, 11, 18)
    val myTimeZone = TimeZone.UTC
    val startOfToday = todaysDate.atStartOfDayIn(myTimeZone)
    println("Start of the day November 18th, 2023 in UTC is: $startOfToday")
}
```
```

> ```kotlin
> import kotlinx.datetime.*
> 
> fun main() {
>     val todaysDate = LocalDate(2023, 11, 18)
>     val myTimeZone = TimeZone.UTC
>     val startOfToday = todaysDate.atStartOfDayIn(myTimeZone)
>     println("Start of the day November 18th, 2023 in UTC is: $startOfToday")
> }

Reviewer: ```suggestion
```kotlin
import kotlinx.datetime.*

fun main() {
    val todaysDate = LocalDate(2023, 11, 18)
    val myTimeZone = TimeZone.UTC
    val startOfToday = todaysDate.atStartOfDayIn(myTimeZone)
    println("Start of the day November 18th, 2023 in UTC is: $startOfToday")
}
```
```

> ```kotlin
> import kotlinx.datetime.*
> 
> fun main() {
>     val todaysDate = LocalDate(2023, 11, 18)
>     val myTimeZone = TimeZone.UTC
>     val startOfToday = todaysDate.atStartOfDayIn(myTimeZone)
>     println("Start of the day November 18th, 2023 in UTC is: $startOfToday")
> }

Reviewer: ```suggestion
```kotlin
import kotlinx.datetime.*

fun main() {
    val todaysDate = LocalDate(2023, 11, 18)
    val myTimeZone = TimeZone.UTC
    val startOfToday = todaysDate.atStartOfDayIn(myTimeZone)
    println("Start of the day November 18th, 2023 in UTC is: $startOfToday")
}

```
```

> }

Reviewer: Add the closing backticks and leave no space after the last line of the code.

Author: ok thank you

> 

Reviewer: ```suggestion
```

```

---
Title: '.atStartOfDayIn()'
Description: 'Finds an instant that corresponds to the start of a date in a particular time zone.'
Subjects:
  - 'Computer Science'
  - 'Code Foundations'
Tags:
  - 'Functions'
  - 'Data Structures'
  - 'Time'
  - 'Date'
CatalogContent:
  - 'learn-kotlin'
  - 'paths/computer-science'
---

> The **`.atStartOfDayIn()`** function is used to obtain the start of the day `LocalDateTime` in a specific time zone. It returns an `Instant` (an instantaneous point in time. It is like a super-accurate clock in Kotlin which helps us pinpoint the exact moment when something happens in our code) that corresponds to the start of a date in a particular time zone.

Reviewer 2: ```suggestion
The **`.atStartOfDayIn()`** function obtains the start of the day `LocalDateTime` in a specific time zone. It returns an `Instant` (an instantaneous point in time. This serves as a highly accurate clock in Kotlin, aiding in pinpointing the exact moment when actions occur in code) corresponding to the start of a date in a particular time zone.
```
Avoid using first and second-person pronouns, i.e. "I", "we", and "you".

## Syntax

```pseudo
fun LocalDate.atStartOfDayIn(TimeZone): Instant
```

> - `LocalDate`: represents a date without reference to a particular time zone.

Reviewer 2: ```suggestion
- `LocalDate`: This represents a date without any reference to a specific time zone.
```

> - `TimeZone`: represents a time zone in which date and time calculations are performed.

Reviewer 2: ```suggestion
- `TimeZone`: This represents a time zone in which date and time calculations are performed.
```

> - `Instant`: a moment in time.

Reviewer 2: ```suggestion
- `Instant`: This is a moment in time. It's a point on the time scale that captures the exact time something happens without regard to time zones or local calendars.
```

## Example

> The following example demonstrates how to use `.atStartOfDayIn`.

Reviewer 2: ```suggestion
The following example demonstrates how to use `.atStartOfDayIn`:
```

```kotlin
import kotlinx.datetime.*

fun main() {
    val todaysDate = LocalDate(2023, 11, 18)
    val myTimeZone = TimeZone.UTC
    val startOfToday = todaysDate.atStartOfDayIn(myTimeZone)
    println("Start of the day November 18th, 2023 in UTC is: $startOfToday")
}
```

> The output of the example above is:

Reviewer 2: ```suggestion
The above example will result in the following output:
```

---
Title: '.atStartOfDayIn()'
Description: 'Finds an instant that corresponds to the start of a date in a particular time zone.'
Subjects:
  - 'Computer Science'
  - 'Code Foundations'
Tags:
  - 'Functions'
  - 'Data Structures'
  - 'Time'
  - 'Date'
CatalogContent:
  - 'learn-kotlin'
  - 'paths/computer-science'
---

The **`.atStartOfDayIn()`** function is used to obtain the start of the day `LocalDateTime` in a specific time zone. It returns an `Instant` (an instantaneous point in time. It is like a super-accurate clock in Kotlin which helps us pinpoint the exact moment when something happens in our code) that corresponds to the start of a date in a particular time zone.

## Syntax

```pseudo
fun LocalDate.atStartOfDayIn(TimeZone): Instant
```

- `LocalDate`: represents a date without reference to a particular time zone.
- `TimeZone`: represents a time zone in which date and time calculations are performed.
- `Instant`: a moment in time.

## Example

The following example demonstrates how to use `.atStartOfDayIn`.

> ```kotlin
> import kotlinx.datetime.*
> 
> fun main() {
>     val todaysDate = LocalDate(2023, 11, 18)
>     val myTimeZone = TimeZone.UTC
>     val startOfToday = todaysDate.atStartOfDayIn(myTimeZone)
>     println("Start of the day November 18th, 2023 in UTC is: $startOfToday")
> }
> 
> ```

Reviewer: ```suggestion
```kotlin
import kotlinx.datetime.*

fun main() {
    val todaysDate = LocalDate(2023, 11, 18)
    val myTimeZone = TimeZone.UTC
    val startOfToday = todaysDate.atStartOfDayIn(myTimeZone)
    println("Start of the day November 18th, 2023 in UTC is: $startOfToday")
}
```
```

---
Title: '.atStartOfDayIn()'
Description: 'Finds an instant that corresponds to the start of a date in a particular time zone.'
Subjects:
  - 'Computer Science'
  - 'Code Foundations'
Tags:
  - 'Functions'
  - 'Data Structures'
  - 'Time'
  - 'Date'
CatalogContent:
  - 'learn-kotlin'
  - 'paths/computer-science'
---

The **`.atStartOfDayIn()`** function is used to obtain the start of the day `LocalDateTime` in a specific time zone. It returns an `Instant` (an instantaneous point in time. It is like a super-accurate clock in Kotlin which helps us pinpoint the exact moment when something happens in our code) that corresponds to the start of a date in a particular time zone.

## Syntax

```pseudo
fun LocalDate.atStartOfDayIn(TimeZone): Instant
```

- `LocalDate`: represents a date without reference to a particular time zone.
- `TimeZone`: represents a time zone in which date and time calculations are performed.
- `Instant`: a moment in time.

## Example

> The following example demonstrates how to use **`.atStartOfDayIn`**.

Reviewer: ```suggestion
The following example demonstrates how to use `.atStartOfDayIn`.
```
Bold only the first occurrence of the method name.

> ```kotlin
> import kotlinx.datetime.LocalDate
> import kotlinx.datetime.TimeZone
> 
> fun main() {
>     val localDate = LocalDate(2023, 11, 9)
>     val timeZone = TimeZone.UTC
>     val startOfDayDateTime = localDate.atStartOfDayIn(timeZone)
>     println("Start of the day November 9th, 2023 in UTC is: $startOfDayDateTime")
> }
> 
> ```

Reviewer: ```suggestion
```kotlin
import kotlinx.datetime.*

fun main() {
    val localDate = LocalDate(2023, 11, 9)
    val timeZone = TimeZone.UTC
    val startOfDayDateTime = localDate.atStartOfDayIn(timeZone)
    println("Start of the day November 9th, 2023 in UTC is: $startOfDayDateTime")
}
```
```
1. You can directly use the `import kotlinx.datetime.*` to import all methods from the library, Use this instead of adding methods one by one. 
2. Change the variable names as they are very similar to some of the reserved words of Kotlin.

---
Title: '.atStartOfDayIn()'
Description: 'Finds an instant that corresponds to the start of a date in a particular time zone.'
Subjects:
  - 'Computer Science'
  - 'Code Foundations'
Tags:
  - 'Functions'
  - 'Data Structures'
  - 'Time'
  - 'Date'
CatalogContent:
  - 'learn-kotlin'
  - 'paths/computer-science'
---

> The **`.atStartOfDayIn()`** function is used to obtain the start of the day `LocalDateTime` in a specific time zone. It returns an `Instant` (a moment in time, which is a date and time representation that provides a point in time on the global timeline, usually measured in seconds and nanoseconds) that corresponds to the start of a date in a particular time zone.

Reviewer: ```suggestion
The **`.atStartOfDayIn()`** function is used to obtain the start of the day `LocalDateTime` in a specific time zone. It returns an `Instant` (a moment in time, which is a date and time representation that provides a point in time on the global timeline, usually measured in seconds and nanoseconds) that corresponds to the start of a date in a particular time zone.
```
Try to simplify the part in which you are explaining the "Instant".

## Syntax

```pseudo
fun LocalDate.atStartOfDayIn(TimeZone): Instant
```

> - `LocalDate`: represents a date without reference to a particular time zone
> - `TimeZone`: represents a time zone in which date and time calculations are performed
> - `Instant`: a moment in time

Reviewer: ```suggestion
- `LocalDate`: represents a date without reference to a particular time zone.
- `TimeZone`: represents a time zone in which date and time calculations are performed.
- `Instant`: a moment in time.
```

## Example

This example demonstrates how to use **`.atStartOfDayIn`**.

```kotlin

> import kotlinx.datetime.LocalDate
> import kotlinx.datetime.TimeZone
> 
> fun main() {
>     val localDate = LocalDate(2023, 11, 9)
> 
>     val timeZone = TimeZone.UTC
> 
>     val startOfDayDateTime = localDate.atStartOfDayIn(timeZone)
> 
>     println("Start of the day November 9th, 2023 in UTC is: $startOfDayDateTime")
> }

Reviewer: ```suggestion
import kotlinx.datetime.LocalDate
import kotlinx.datetime.TimeZone

fun main() {
    val localDate = LocalDate(2023, 11, 9)
    val timeZone = TimeZone.UTC
    val startOfDayDateTime = localDate.atStartOfDayIn(timeZone)
    println("Start of the day November 9th, 2023 in UTC is: $startOfDayDateTime")
}
```
Remove the unnecessary line spaces from the code.

```

> The output will be similar to the following:

Reviewer: Simply rewrite this line as it contains some plagiarism.
